### ISSUE TITLE
我好像没有找到注意力机制中的con这个参数

### ISSUE THREAD

**Issue by Vzoooong** [jqtangust/IUF#6]
您的论文中：
![image](https://github.com/user-attachments/assets/90522342-f265-4852-b69b-77c35cad1095)
但是您的代码：
class MultiHeadSelfAttention(nn.Module):
    def __init__(self, dim, num_heads):
        super(MultiHeadSelfAttention, self).__init__()
        self.dim = dim
        self.num_heads = num_heads
        self.head_dim = dim // num_heads

        assert dim % self.num_heads == 0, "embedding dimension must be divisible by number of heads"

        self.query = nn.Linear(dim, dim)
        self.key = nn.Linear(dim, dim)
        self.value = nn.Linear(dim, dim)
        self.out = nn.Linear(dim, dim)

    def forward(self, x):
        b, h, w, _ = x.size()
        
        q = self.query(x)
        k = self.key(x)
        v = self.value(x)

        q = rearrange(q, 'b h w (n d) -> b n h w d', n=self.num_heads)
        k = rearrange(k, 'b h w (n d) -> b n h w d', n=self.num_heads)
        v = rearrange(v, 'b h w (n d) -> b n h w d', n=self.num_heads)

        k_T = k.transpose(-2, -1)
        scores = torch.matmul(q, k_T) / self.head_dim**0.5
        attention = F.softmax(scores, dim=-1)
        out = torch.matmul(attention, v)

        out = rearrange(out, 'b n h w d -> b h w (n d)')
        out = self.out(out)
        return out
        请问这个con参数在哪里呢？

**Comment by jqtangust** [comment]
这部分的out作为Con参数，提供给reconstruction network
